pajbot/web/routes/base/login.py

```diff
@@ -104,25 +104,24 @@ def authorized():
             return redirect(next_url)
         session["twitch_token"] = (resp["access_token"],)
 
-        me_api_response = twitch.get("users")
-        if len(me_api_response.data["data"]) < 1:
+        token = UserAccessToken.from_api_response(resp)
+
+        me_api_response = app.twitch_helix_api.get_user_data_by_client_id_and_access_token(
+            app.bot_config["twitchapi"]["client_id"], token.access_token
+        )
+        if me_api_response is None:
             return render_template("login_error.html")
 
         with DBManager.create_session_scope(expire_on_commit=False) as db_session:
             me = User.from_basics(
-                db_session,
-                UserBasics(
-                    me_api_response.data["data"][0]["id"],
-                    me_api_response.data["data"][0]["login"],
-                    me_api_response.data["data"][0]["display_name"],
-                ),
+                db_session, UserBasics(me_api_response["id"], me_api_response["login"], me_api_response["display_name"])
```

Comment:
to avoid this, the helix API wrapper could get a method that returns UserBasics given some authorization

(but I would see it as optional)